cli/src/commit_templater.rs

```diff
@@ -568,11 +568,72 @@ fn builtin_commit_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, Comm
             Ok(L::wrap_boolean(out_property))
         },
     );
+    map.insert(
+        "bookmarks",
+        |language, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let index = language
+                .keyword_cache
+                .bookmarks_index(language.repo)
+                .clone();
+            let out_property = self_property.map(move |commit| {
+                index
+                    .get(commit.id())
+                    .iter()
+                    .filter(|ref_name| ref_name.is_local() || !ref_name.synced)
+                    .cloned()
+                    .collect()
+            });
+            Ok(L::wrap_ref_name_list(out_property))
+        },
+    );
+    map.insert(
+        "local_bookmarks",
+        |language, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let index = language
+                .keyword_cache
+                .bookmarks_index(language.repo)
+                .clone();
+            let out_property = self_property.map(move |commit| {
+                index
+                    .get(commit.id())
+                    .iter()
+                    .filter(|ref_name| ref_name.is_local())
+                    .cloned()
+                    .collect()
+            });
+            Ok(L::wrap_ref_name_list(out_property))
+        },
+    );
+    map.insert(
+        "remote_bookmarks",
+        |language, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let index = language
+                .keyword_cache
+                .bookmarks_index(language.repo)
+                .clone();
+            let out_property = self_property.map(move |commit| {
+                index
+                    .get(commit.id())
+                    .iter()
+                    .filter(|ref_name| ref_name.is_remote())
+                    .cloned()
+                    .collect()
+            });
+            Ok(L::wrap_ref_name_list(out_property))
+        },
+    );
+    // TODO: Remove the following block after jj 0.27+
```

Comment:
nit: you can copy `map["bookmarks"]`, etc. instead of duplicating the function bodies.

(also applies to revset function table.)

Reply:
done